### ISSUE TITLE
Some files are marked as blocked

### ISSUE THREAD

**Issue by jonnadulasudhakar** [cyberark/CYBRHardeningCheck#46]
## Summary
When trying to execute main.ps1 , I am getting the error message some files are blocked.

## Steps to Reproduce

Even when executing with Verbose mode, got the same error. 

## Actual Results (including error logs, if applicable)
A clear and concise description of what actually did happen.

## Reproducible
   * [ ] Always 
   * [ ] Sometimes
   * [ ] Non-Reproducible
   
## Version/Tag number
What version of the product are you running? Any version info that you can share is helpful. 
For example, you might give the version from Docker logs, the Docker tag, a specific download URL, 
the output of the `/info` route, etc.

## Environment setup
Can you describe the environment in which this product is running? Is it running on a VM / in a container / in a cloud? 
Which cloud provider? Which container orchestrator (including version)? 
The more info you can share about your runtime environment, the better we may be able to reproduce the issue.

**Comment by jonnadulasudhakar** [comment]
PS C:\Temp\CYBRHardeningCheck-main\CYBRHardeningCheck-main\CYBRHardeningCheck> .\Main.ps1
=======================================
Starting script (v2.8.2)
Some files are marked as blocked
To solve this you can run the following command: Get-ChildItem -Path C:\Temp\CYBRHardeningCheck-main\CYBRHardeningCheck-main\CYBRHardeningCheck -Recurse | Unblock-File
Script ended
PS C:\Temp\CYBRHardeningCheck-main\CYBRHardeningCheck-main\CYBRHardeningCheck> ./Main.ps1 -Debug -Verbose
VERBOSE: Removing the imported "Write-LogMessage" function.
VERBOSE: Removing the imported "Test-ServiceRunWithLocalUser" function.
VERBOSE: Removing the imported "Test-LocalUser" function.
VERBOSE: Removing the imported "Test-InstalledWindowsRole" function.
VERBOSE: Removing the imported "Test-InDomain" function.
VERBOSE: Removing the imported "Test-CurrentUserLocalAdmin" function.
VERBOSE: Removing the imported "Test-CredFileVerificationType" function.
VERBOSE: Removing the imported "Start-HardeningSteps" function.
VERBOSE: Removing the imported "Set-CurrentComponentFolderPath" function.
VERBOSE: Removing the imported "Join-ExceptionMessage" function.
VERBOSE: Removing the imported "Get-WMIItem" function.
VERBOSE: Removing the imported "Get-ServiceInstallPath" function.
VERBOSE: Removing the imported "Get-SeceditAnalysisResults" function.
VERBOSE: Removing the imported "Get-ParsedFileNameByOS" function.
VERBOSE: Removing the imported "Get-ParsedFileNameByComponent" function.
VERBOSE: Removing the imported "Get-LocalSystem" function.
VERBOSE: Removing the imported "Get-LocalAdministrators" function.
VERBOSE: Removing the imported "Get-DnsHost" function.
VERBOSE: Removing the imported "Get-DetectedComponents" function.
VERBOSE: Removing the imported "Get-CurrentComponentFolderPath" function.
VERBOSE: Removing the imported "ConvertTo-Bool" function.
VERBOSE: Removing the imported "Convert-SIDToName" function.
VERBOSE: Removing the imported "Convert-NameToSID" function.
VERBOSE: Removing the imported "Compare-UserRight" function.
VERBOSE: Removing the imported "Compare-UserPermissions" function.
VERBOSE: Removing the imported "Compare-UserFlags" function.
VERBOSE: Removing the imported "Compare-ServiceStatus" function.
VERBOSE: Removing the imported "Compare-RegistryValue" function.
VERBOSE: Removing the imported "Compare-PolicyEntry" function.
VERBOSE: Removing the imported "Compare-EventLogSizeAndRetentionSettings" function.
VERBOSE: Removing the imported "Compare-AuditRulesFromPath" function.
VERBOSE: Removing the imported "Compare-AmountOfUserPermissions" function.
VERBOSE: Removing the imported "Compare-AdvancedAuditPolicySubCategory" function.
VERBOSE: Loading module from path 'C:\Temp\CYBRHardeningCheck-main\CYBRHardeningCheck-main\CYBRHardeningCheck\bin\CommonUtil.psm1'.
VERBOSE: Importing function 'Compare-AdvancedAuditPolicySubCategory'.
VERBOSE: Importing function 'Compare-AmountOfUserPermissions'.
VERBOSE: Importing function 'Compare-AuditRulesFromPath'.
VERBOSE: Importing function 'Compare-EventLogSizeAndRetentionSettings'.
VERBOSE: Importing function 'Compare-PolicyEntry'.
VERBOSE: Importing function 'Compare-RegistryValue'.
VERBOSE: Importing function 'Compare-ServiceStatus'.
VERBOSE: Importing function 'Compare-UserFlags'.
VERBOSE: Importing function 'Compare-UserPermissions'.
VERBOSE: Importing function 'Compare-UserRight'.
VERBOSE: Importing function 'Convert-NameToSID'.
VERBOSE: Importing function 'Convert-SIDToName'.
VERBOSE: Importing function 'ConvertTo-Bool'.
VERBOSE: Importing function 'Get-CurrentComponentFolderPath'.
VERBOSE: Importing function 'Get-DetectedComponents'.
VERBOSE: Importing function 'Get-DnsHost'.
VERBOSE: Importing function 'Get-LocalAdministrators'.
VERBOSE: Importing function 'Get-LocalSystem'.
VERBOSE: Importing function 'Get-ParsedFileNameByComponent'.
VERBOSE: Importing function 'Get-ParsedFileNameByOS'.
VERBOSE: Importing function 'Get-SeceditAnalysisResults'.
VERBOSE: Importing function 'Get-ServiceInstallPath'.
VERBOSE: Importing function 'Get-WMIItem'.
VERBOSE: Importing function 'Join-ExceptionMessage'.
VERBOSE: Importing function 'Set-CurrentComponentFolderPath'.
VERBOSE: Importing function 'Start-HardeningSteps'.
VERBOSE: Importing function 'Test-CredFileVerificationType'.
VERBOSE: Importing function 'Test-CurrentUserLocalAdmin'.
VERBOSE: Importing function 'Test-InDomain'.
VERBOSE: Importing function 'Test-InstalledWindowsRole'.
VERBOSE: Importing function 'Test-LocalUser'.
VERBOSE: Importing function 'Test-ServiceRunWithLocalUser'.
VERBOSE: Importing function 'Write-LogMessage'.
VERBOSE: Removing the imported "ValidateServerRoles" function.
VERBOSE: Removing the imported "RemoteDesktopServices" function.
VERBOSE: Removing the imported "RegistryPermissions" function.
VERBOSE: Removing the imported "RegistryAudits" function.
VERBOSE: Removing the imported "ImportingINFConfiguration" function.
VERBOSE: Removing the imported "FileSystemPermissions" function.
VERBOSE: Removing the imported "FileSystemAudit" function.
VERBOSE: Removing the imported "EventLogSizeAndRetention" function.
VERBOSE: Removing the imported "DisableServices" function.
VERBOSE: Removing the imported "DisableScreenSaver" function.
VERBOSE: Removing the imported "AdvancedAuditPolicyConfiguration" function.
VERBOSE: Loading module from path 'C:\Temp\CYBRHardeningCheck-main\CYBRHardeningCheck-main\CYBRHardeningCheck\bin\GeneralHardeningSteps.psm1'.
VERBOSE: Exporting function 'ImportingINFConfiguration'.
VERBOSE: Exporting function 'ValidateServerRoles'.
VERBOSE: Exporting function 'DisableScreenSaver'.
VERBOSE: Exporting function 'AdvancedAuditPolicyConfiguration'.
VERBOSE: Exporting function 'RemoteDesktopServices'.
VERBOSE: Exporting function 'EventLogSizeAndRetention'.
VERBOSE: Exporting function 'RegistryAudits'.
VERBOSE: Exporting function 'RegistryPermissions'.
VERBOSE: Exporting function 'FileSystemPermissions'.
VERBOSE: Exporting function 'FileSystemAudit'.
VERBOSE: Exporting function 'DisableServices'.
VERBOSE: Importing function 'AdvancedAuditPolicyConfiguration'.
VERBOSE: Importing function 'DisableScreenSaver'.
VERBOSE: Importing function 'DisableServices'.
VERBOSE: Importing function 'EventLogSizeAndRetention'.
VERBOSE: Importing function 'FileSystemAudit'.
VERBOSE: Importing function 'FileSystemPermissions'.
VERBOSE: Importing function 'ImportingINFConfiguration'.
VERBOSE: Importing function 'RegistryAudits'.
VERBOSE: Importing function 'RegistryPermissions'.
VERBOSE: Importing function 'RemoteDesktopServices'.
VERBOSE: Importing function 'ValidateServerRoles'.
=======================================
Starting script (v2.8.2)
Running in Debug Mode
Running in Verbose Mode
Some files are marked as blocked
To solve this you can run the following command: Get-ChildItem -Path C:\Temp\CYBRHardeningCheck-main\CYBRHardeningCheck-main\CYBRHardeningCheck -Recurse | Unblock-File
Script ended
PS C:\Temp\CYBRHardeningCheck-main\CYBRHardeningCheck-main\CYBRHardeningCheck> Get-ChildItem -Path C:\Temp\CYBRHardeningCheck-main\CYBRHardeningCheck-main\CYBRHardeningCheck -Recurse | Unblock
-File
PS C:\Temp\CYBRHardeningCheck-main\CYBRHardeningCheck-main\CYBRHardeningCheck>

**Comment by jonnadulasudhakar** [comment]
Any changes ? Do we need to download new file ?

**Comment by AssafMiron** [comment]
Hi @jonnadulasudhakar ,
Sorry, this was closed automatically and I forgot to comment 
Yes, the main.ps1 file was changed and should include a fix to this issue
you can download only this file and replace in your local machine

otherwise, you can change line 350 in the main.ps1
from: 
# Check if relevant files are blocked
If($null -eq $(Get-ChildItem -Path $ScriptLocation -Include ('*.ps1','*.psm1','*.dll') -Recurse | Get-Item -Stream “Zone.Identifier” -ErrorAction SilentlyContinue))

to:
# Check if relevant files are blocked
If($null -ne $(Get-ChildItem -Path $ScriptLocation -Include ('*.ps1','*.psm1','*.dll') -Recurse | Get-Item -Stream “Zone.Identifier” -ErrorAction SilentlyContinue))

Regards,
Assaf

**Comment by jonnadulasudhakar** [comment]
Thanks for the update. After making the changes to line 350, I am getting below error:

VERBOSE: Importing function 'ValidateServerRoles'.

Cannot write message

At C:\Temp\CYBRHardeningCheck-main\CYBRHardeningCheck-main\CYBRHardeningCheck\bin\CommonUtil.psm1:130 char:3

+         Throw $(New-Object System.Exception ("Cannot write message"), ...

+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

    + CategoryInfo          : OperationStopped: (:) [], Exception

    + FullyQualifiedErrorId : Cannot write message

 Please advise 

Thanks,

**Comment by AssafMiron** [comment]
Thanks for the info @jonnadulasudhakar
I recommend running the script with verbose logging (for example: main.ps1
-Verbose)
That will create a log file that should have additional info on why this
fail
Since the original issue was solved, could you please open a new issue for
this and post the result of the verbose logging?

One additional thing I would check is if the file CommonUtil.psm1 in the
bin folder is blocked - Go to properties and see if you have a checkbox of
Unblock file
If you do, please run the command:
Get-ChildItem -Path
C:\Temp\CYBRHardeningCheck-main\CYBRHardeningCheck-main\CYBRHardeningCheck
-Recurse | Unblock-File
Thanks,
Assaf
